foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConversionManager.java

```diff
@@ -75,6 +84,57 @@
     /** Store the list of Classes that can be converted from to the key. */
     protected Hashtable dataTypesConvertedToAClass;
 
+    private static ZoneId getDefaultZoneOffset() {
+        if (defaultZoneOffset == null) {
+            ZoneId tzoneid = null;
+            String tzone = null;
+            try {
+                tzone = PrivilegedAccessHelper.getSystemProperty(SystemProperties.CONVERSION_USE_TIMEZONE);
+                if (tzone != null) {
+                    try {
+                        tzoneid = java.time.ZoneId.of(tzone);
+                    } catch (Throwable t) {
+                        // If an invalid time zone id is supplied, then fall back to checking for checking for using
+                        // either UTC or the system's default time zone.
+                        if (AbstractSessionLog.getLog().shouldLog(SessionLog.WARNING)) {
```

Comment:
I agree it's better to stay safe but do we really need to do this check 3 times in a row? :-)

...there's the same check in [the log method itself](https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/logging/AbstractSessionLog.java#L397) as well as in the method [that log method calls](https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/logging/AbstractSessionLog.java#L575)

Reply:
The first catch-block is for catching a conversion failure with java.time.ZoneId.of(tzone), for which I wanted to log a specific warning:

```
                    try {
                        tzoneid = java.time.ZoneId.of(tzone);
                    } catch (Throwable t) {
                        // If an invalid time zone id is supplied, then fall back to checking for checking for using
                        // either UTC or the system's default time zone.
                        if (AbstractSessionLog.getLog().shouldLog(SessionLog.WARNING)) {
                            AbstractSessionLog.getLog().log(SessionLog.WARNING, "invalid_tzone", 
                                    SystemProperties.CONVERSION_USE_TIMEZONE, tzone);
                        }
                    }
```

The second catch block, surrounding that, is a guard against an (unlikely) exception occurring elsewhere, like say 
```
tzone = PrivilegedAccessHelper.getSystemProperty(SystemProperties.CONVERSION_USE_TIMEZONE);
```
Which is just swallowed.

If either of the above cases threw an Exception, it would fall back to processing `SystemProperties.CONVERSION_USE_DEFAULT_TIMEZONE`

Which has its own Exception catch block, which emits a warning specific to the attempt to consume that specific property.

So yes, it needed that many catches to ensure that both a proper warning message is emitted and that the follow up property is attempted.